sw/device/silicon_creator/rom_ext/doc/rescue.md

```diff
@@ -0,0 +1,143 @@
+# ROM_EXT Rescue Protocol
+
+## Introduction
+
+The ROM_EXT rescue protocol permits the chip owner to perform device diagnostic and management actions when there is no functional application firmware.
+
+These actions include:
+- Flashing a new application image.
+- Interatcting with Boot Services.
+- Requesting the Boot Log data.
+- Performing Ownership Transfer.
+
+Currently, the rescue protocol uses the console UART as the main communication channel.
+In the future, rescue operations may be available via additional interfaces.
+
+## Basic Operation
+
+In order to interact with the rescue protocol, the user needs to signal to the ROM_EXT that rescue a rescue operation is desired.
+
+Once signalled, the ROM_EXT will acknowledge entry into rescue mode and the user can then specify a specific rescue action and either upload or download any data block associated with that action.
+
+Rescue mode is exited via either an in-band or out-of-band reset signal.
+
+## Serial Rescue Protocol
+
+### Requesting Rescue
+
+In order to signal to the ROM_EXT that a rescue operation is desired, the user should assert the serial break condition for at least 350us while the ROM_EXT boots.
+When the ROM_EXT detects the serial break condition, it will respond with an acknowledgement:
+
+```
+rescue: remember to clear break
+```
+
+Upon entry into rescue mode, the user should clear the serial break condition to allow further interaction with rescue mode.
+
+### Rescue Operations
+
+By default, rescue will enter firmware rescue mode and will prompt the user to upload a new firmware image via the Xmodem-CRC protocol.
```

Comment:
I think we talked about a timeout, such that if 100ms pass after entering rescue mode, without any command/data being sent, that normal boot would proceed.  (This will prevent a loose screw in a Chromebook having short-circuited the UART RX pin from rendering the device useless to the end user, and instead add just 0.1 second to the boot time.)  We can debate where merely removing the break condition, without sending any actual serial data, should in itself be enough to cancel the 100ms timeout.

Reply:
We did discuss having a timeout.  I intend to make the timeout an owner-configurable feature so that owners can set the value according to their own needs.